### DISCUSSION TITLE
How to configure Orbot through ADB on rooted device?

### DISCUSSION THREAD

**Discussion by a-t-0** [guardianproject/orbot-android#775] [Q&A]
Thank you for your marvelous work on this app! I'm automating a calendar setup over tor, and was looking for an open source, and more robust way to automatically configure Orbot to:
0. Allow the `Connection request` (prompt: `ok`)
1. Select the `Davx5` application to be "`torified`".
2. Start the Orbot service.

My current approach is to use simulated keypresses, as described [here](https://android.stackexchange.com/a/25670/254078). However, if a call comes in, or messages popup, this may lead to unintended effects. 

So I thought of two options: 
0. Determine where the Orbot app settings are stored on the phone and modify those.
1. Pre-configure a build with the intended settings.
2. Write a termux script that performs the Orbot configuration settings.

However, the precompilation is somewhat involved, especially to do on arbitrary pc's that just want the calendar, and if I do it myself (once), users have to rely on my compiled file, which I prefer to avoid.

Furthermore, I did not yet find if/where the Orbot app settings/configurations are stored on a device, and whether this location is the same for each device/automatically retrievable.

Do you have suggestions on how I can robustly and automatically configure Orbot through ADB?

**Comment by n8fr8** [comment]
Not sure we want to officially support configuration through ADB, and given you would need root access to modify Orbot config files, not sure it is possible.

Perhaps though we can add support to Orbot for an API to support torification?
